aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -145,169 +143,86 @@ public Orchestrator(
+            LOG.debug("Lock released.");
+        });
     }
 
-    private Completable transitionCompletable() {
-        Mode current = currentMode.get();
-        Mode target = targetMode.get();
-        if (ObjectsCompat.equals(current, target)) {
-            return Completable.complete();
-        }
-        LOG.info(String.format(Locale.US,
-            "DataStore orchestrator transitioning states. " +
-                "Current mode = %s, target mode = %s.", current, target
-        ));
-
-        switch (target) {
+    private void transitionToStarted() throws DataStoreException {
+        switch (currentMode.get()) {
             case STOPPED:
-                return transitionToStopped(current);
+                LOG.info("Starting the orchestrator.");
+                startObservingStorageChanges();
+                if (apiSyncEnabled) {
+                    startApiSync();
+                } else {
+                    LOG.info("Started the orchestrator in local only mode.");
+                    publishReadyEvent();
+                }
+                break;
             case LOCAL_ONLY:
-                return transitionToLocalOnly(current);
+                if (apiSyncEnabled) {
+                    startApiSync();
+                }
+                break;
+            case SYNC_STARTING:
             case SYNC_VIA_API:
-                return transitionToApiSync(current);
             default:
-                return unknownMode(target);
+                break;
         }
     }
 
-    /**
-     * Stop the orchestrator.
-     * @return A completable which emits success when orchestrator stops
-     */
-    public synchronized Completable stop() {
+    private Completable transitionToStopped() {
         LOG.info("DataStore orchestrator stopping. Current mode = " + currentMode.get().name());
-        if (tryAcquireStartStopLock(LOCAL_OP_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
-            disposables.clear();
-            return transitionToStopped(currentMode.get())
-                .subscribeOn(startStopScheduler)
-                .doFinally(startStopSemaphore::release);
-        } else {
-            return Completable.error(new DataStoreException("Unable to acquire orchestrator lock. " +
-                                                                "Transition currently in progress.",
-                                                            "Retry your operation"));
-        }
-
-    }
-
-    private static Completable unknownMode(Mode mode) {
-        return Completable.error(new DataStoreException(
-            "Orchestrator state machine made reference to unknown mode = " + mode.name(),
-            AmplifyException.REPORT_BUG_TO_AWS_SUGGESTION
-        ));
-    }
-
-    private Completable transitionToStopped(Mode current) {
-        switch (current) {
+        disposables.clear();
+        switch (currentMode.get()) {
             case SYNC_VIA_API:
+            case SYNC_STARTING:
```

Comment:
I did consider adding the other 3 transitions:
 - `STOPPED` -> `LOCAL_ONLY`
 - `SYNC_VIA_API` -> `LOCAL_ONLY`
 - `LOCAL_ONLY` -> `STOPPED` 

Ultimately I didn't because all of these transitions are synchronized with the semaphore, so there's no benefit of tracking the state.  

I did have to add `SYNC_STARTING` (`LOCAL_ONLY` -> `SYNC_VIA_API`), since this transition is no longer locked with the semaphore, to ensure the API sync wouldn't ever be started multiple times.

Reply:
Looking closer at this, when the state was `SYNC_STARTING` or `SYNC_VIA_API`, I was always doing the same thing, so I don't actually need both.   I've gotten rid of `SYNC_STARTING`, and just updated logic to set the state to SYNC_VIA_API as soon as it starts to sync, rather than at the end of sync.  That way, it will still prevent sync from starting more than once.